src/components/pages/lessons/LessonInfo.tsx

```diff
@@ -47,6 +48,71 @@ const NextUp: FunctionComponent<NextUpProps> = ({data, currentLessonSlug}) => {
   ) : null
 }
 
+const TweetLink: FunctionComponent<{
+  source: {
+    title: string
+    slug: string
+  }
+  className?: string
+}> = ({source, className = ''}) => {
+  if (source) {
+    return (
+      <a
+        className={`flex items-center rounded p-2 bg-gray-200 hover:bg-gray-400 transition-colors ease-in-out duration-150 ${className}`}
+        target="_blank"
+        rel="noopener noreferrer"
+        href={`https://twitter.com/intent/tweet/?text=${encodeURIComponent(
+          source.title + ', lesson by @eggheadio',
+        )}&url=${encodeURIComponent(
+          `https://next.egghead.io/lessons/${source.slug}/`,
```

Comment:
use `NEXT_PUBLIC_REDIRECT_URI` or similar from ENV instead of hardcoding it?

Reply:
For same domain if you are URL building `/` works

Does source have a `path`?

Reply:
yes, it has

Reply:
but seems we need `process.env.NEXT_PUBLIC_REDIRECT_URI` anyway